Source/Contrib/ContentManager/Models/Consist.cs

```diff
@@ -75,18 +75,51 @@ public Consist(Content content)
                         MinCouplerStrengthN = Math.Min(MinCouplerStrengthN, wagonFile.MinCouplerStrengthN);
                         if (wagonFile.MaxBrakeForceN > 0) { NumOperativeBrakes++; }
 
-                        if (wag.IsEngine && engFile.MaxForceN > 25000)  // exclude legacy driving trailers / cab-cars
+                        if (wag.IsEngine)
                         {
-                            EngCount++;
+                            // see MSTSLocomotive.Initialize()
+                            ortsEngAxles = engFile.NumDriveAxles;
+                            if (ortsEngAxles >= 0) { numDriveAxles = ortsEngAxles; }
+                            else if (engFile.NumEngWheels > 7f) { numDriveAxles = (int)(engFile.NumEngWheels / 2f); }
+                            else if (engFile.NumEngWheels > 0) { numDriveAxles = (int)engFile.NumEngWheels; }
+                            else { numDriveAxles = 4; }
 
-                            MaxPowerW += engFile.MaxPowerW;
-                            MaxTractiveForceN += engFile.MaxForceN;
+                            if (engFile.MaxForceN > 25000)  // exclude legacy driving trailers / cab-cars
+                            {
+                                EngCount++;
+                                MaxPowerW += engFile.MaxPowerW;
+                                MaxTractiveForceN += engFile.MaxForceN;
+                                MaxDynamicBrakeForceN += engFile.MaxDynamicBrakeForceN;
+                            }
+                            else { WagCount++; }
                         }
                         else if (!wag.IsEOT && wagonFile.WagonSize.LengthM > 1.1) // exclude legacy EOT
                         {
                             WagCount++;
                         }
-                    } catch (IOException e) // continue without details when eng/wag file does not exist
+
+                        // see MSTSWagon.LoadFromWagFile()
+                        if (ortsEngAxles >= 0 && wagonFile.NumWagAxles >= 0) { numAllAxles = ortsEngAxles + wagonFile.NumWagAxles; }
+                        else if (wagonFile.NumWagAxles >= 0) { numAllAxles =  wagonFile.NumWagAxles; }
+                        else if (wagonFile.NumWagWheels >= 7f) { numAllAxles = (int)(wagonFile.NumWagWheels / 2f); }
+                        else if (wagonFile.NumWagWheels >= 0f) { numAllAxles = (int)wagonFile.NumWagWheels; }
+                        else { numAllAxles = 4; }
+                        if (numDriveAxles > numAllAxles) { numAllAxles = numDriveAxles; }
```

Comment:
This whole section is a bit clunky and also doesn't reflect what the simulator will do. Again, there's no case anywhere in the sim where NumWheels is divided by 2, and despite calculating 'num**All**Axles' most of the cases here will just give the wagon axles. Something like this would be more accurate and concise:

```suggestion
                        if (wagonFile.NumWagAxles != 0 || wag.IsEngine) { numAllAxles = wagonFile.NumWagAxles; }
                        else if (wagonFile.NumWagWheels != 0 && wagonFile.NumWagWheels < 6) { numAllAxles = (int)wagonFile.NumWagWheels; }
                        else { numAllAxles = 4; }
                        
                        numAllAxles += numDriveAxles;
```
In this approach, we are consistent with the (weird) implementation in the simulator, and more consistently produce a correct number of total axles by first calculating the unpowered axles, then adding on powered axles (if there were any).

Reply:
Yes, it is different. Perhaps the thinking is that NumWheels( 6 ) on a wagon could mean a 6 wheel wagon (3 axle stock wasn't unusual in the UK) or a 6 axle wagon, both of which are reasonable, so it's ambiguous and we give up. Meanwhile on a locomotive NumWheels( 6 ) probably means 6 axles as 6 axle locomotives are more prevalent than 3 axle locomotives.

Of course, that is making the assumption NumWheels was set to be the wheels or the axles, and often times it's neither!